flink/src/main/java/org/apache/iceberg/flink/data/vectorized/ConstantColumnVectors.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.data.vectorized;
+
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.table.data.binary.BinaryStringData;
+import org.apache.flink.table.data.vector.BooleanColumnVector;
+import org.apache.flink.table.data.vector.BytesColumnVector;
+import org.apache.flink.table.data.vector.DecimalColumnVector;
+import org.apache.flink.table.data.vector.DoubleColumnVector;
+import org.apache.flink.table.data.vector.FloatColumnVector;
+import org.apache.flink.table.data.vector.IntColumnVector;
+import org.apache.flink.table.data.vector.LongColumnVector;
+import org.apache.flink.table.data.vector.TimestampColumnVector;
+
+class ConstantColumnVectors {
+  private ConstantColumnVectors() {
+  }
+
+  static IntOrcColumnVector ints(Object constant) {
+    return new IntOrcColumnVector(constant);
+  }
+
+  static LongOrcColumnVector longs(Object constant) {
+    return new LongOrcColumnVector(constant);
+  }
+
+  static BooleanOrcColumnVector booleans(Object constant) {
+    return new BooleanOrcColumnVector(constant);
+  }
+
+  static DoubleOrcColumnVector doubles(Object constant) {
+    return new DoubleOrcColumnVector(constant);
+  }
+
+  static FloatOrcColumnVector floats(Object constant) {
+    return new FloatOrcColumnVector(constant);
+  }
+
+  static DecimalOrcColumnVector decimals(Object constant) {
+    return new DecimalOrcColumnVector(constant);
+  }
+
+  static TimestampOrcColumnVector timestamps(Object constant) {
+    return new TimestampOrcColumnVector(constant);
+  }
+
+  static BytesOrcColumnVector bytes(Object constant) {
+    return new BytesOrcColumnVector(constant);
+  }
+
+  private static class IntOrcColumnVector implements IntColumnVector {
+
+    private Object constant;
+
+    private IntOrcColumnVector(Object constant) {
+      this.constant = constant;
+    }
+
+    @Override
+    public boolean isNullAt(int i) {
+      return false;
```

Comment:
If the `constant` instance is null,  then this method should return `true`, right ?   So the correct way is: `return constant == null` here ?

Reply:
the code we get the `idToConstant` is 
```
    Map<Integer, ?> idToConstant = partitionSchema.columns().isEmpty() ? ImmutableMap.of() :
        PartitionUtil.constantsMap(task, RowDataUtil::convertConstant);
```
so I think there should  be no `null` value in `idToConstant `, what do you think ?

Reply:
It's possible that the [constant value to be null ](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java#L47).   Consider that there's an optional field in the partition fields, then if its value is null,  then the constantMap will be null.   The [PartitionUtil#constantsMap](https://github.com/apache/iceberg/blob/90225d6c9413016d611e2ce5eff37db1bc1b4fc5/core/src/main/java/org/apache/iceberg/util/PartitionUtil.java#L47) also clearly stated that there will be null values.

Reply:
ok,I updated it